components/service/glean/src/main/java/mozilla/components/service/glean/error/ErrorRecording.kt

```diff
@@ -0,0 +1,120 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+@file:Suppress("MatchingDeclarationName")
+
+package mozilla.components.service.glean.error
+
+import android.support.annotation.VisibleForTesting
+import kotlinx.coroutines.launch
+import mozilla.components.service.glean.CommonMetricData
+import mozilla.components.service.glean.CounterMetricType
+import mozilla.components.service.glean.Dispatchers
+import mozilla.components.service.glean.Lifetime
+import mozilla.components.service.glean.storages.CountersStorageEngine
+import mozilla.components.support.base.log.logger.Logger
+
+// The labeled counter metrics that store the errors are defined in
+// the `metrics.yaml` for documentation purposes, but are not actually
+// used directly, since the `sendInPings` value needs to match the
+// pings of the metric that is erroring, not some constant value that
+// we could define in `metrics.yaml`.
+
+internal enum class ErrorType {
+    OutOfRange,
+    InvalidType,
```

Comment:
This seems to not be ever used: are we missing some instrumentation? If not, maybe we could remove it and add it later.

Reply:
It was defined in the spec, but I'm happy to remove it for now.  There's also the question as to whether `OutOfRange` and `InvalidValue` are too redundant.  (See the open questions at the top of the PR).

Reply:
Yeah, they feel redundant. +1 on removing that for now, and always using the same.

Reply:
Great.  On this point, I'll merge `OutOfRange` into `InvalidValue` and remove `InvalidType`.